modules/tinymce/src/themes/silver/main/ts/ui/dialog/WindowManager.ts

```diff
@@ -244,6 +246,134 @@ const setup = (extras: WindowManagerSetup): WindowManagerImpl => {
+
+      const dialogInit = {
+        dataValidator,
+        initialData,
+        internalDialog: contents
+      };
+
+      const refreshDocking = () => inlineDialog.on((dialog) => {
+        InlineView.reposition(dialog);
+        Docking.refresh(dialog);
+      });
+
+      const dialogUi = renderInlineDialog<T>(
+        dialogInit,
+        {
+          redial: DialogManager.DialogManager.redial,
+          closeWindow: () => {
+            inlineDialog.on(InlineView.hide);
+            editor.off('ResizeEditor ScrollWindow ElementScroll', refreshDocking);
+            inlineDialog.clear();
+            closeWindow(dialogUi.instanceApi);
+          }
+        },
+        extras.backstages.popup,
+        ariaAttrs
+      );
+
+      const inlineAdditionalBehaviours = (editor: Editor): Behaviour.NamedConfiguredBehaviour<any, any>[] => {
+        return [
+          Docking.config({
+            contextual: {
+              lazyContext: () => Optional.some(Boxes.box(SugarElement.fromDom(editor.getContentAreaContainer()))),
+              fadeInClass: 'tox-dialog-dock-fadein',
+              fadeOutClass: 'tox-dialog-dock-fadeout',
+              transitionClass: 'tox-dialog-dock-transition'
+            },
+            modes: [ 'top', 'bottom' ],
+            lazyViewport: (comp) => {
+              const optScrollingContext = ScrollingContext.detectWhenSplitUiMode(editor, comp.element);
+              return optScrollingContext
+                .map(
+                  (sc) => {
+                    const combinedBounds = ScrollingContext.getBoundsFrom(sc);
+                    return {
+                      bounds: combinedBounds,
+                      optScrollEnv: Optional.some({
+                        currentScrollTop: sc.element.dom.scrollTop,
+                        scrollElmTop: SugarLocation.absolute(sc.element).top
+                      })
+                    };
+                  }
+                ).getOrThunk(
+                  () => {
+                    return {
+                      bounds: Boxes.win(),
+                      optScrollEnv: Optional.none()
+                    };
+                  }
+                );
+            }
+          })
+        ];
+      };
+
+      const inlineDialogComp = GuiFactory.build(InlineView.sketch({
+        lazySink: extras.backstages.popup.shared.getSink,
+        dom: {
+          tag: 'div',
+          classes: [ ]
+        },
+        // Fires the default dismiss event.
+        fireDismissalEventInstead: { },
+        ...isToolbarLocationTop ? { } : { fireRepositionEventInstead: { }},
+        inlineBehaviours: Behaviour.derive([
+          AddEventsBehaviour.config('window-manager-inline-events', [
+            AlloyEvents.run(SystemEvents.dismissRequested(), (_comp, _se) => {
+              AlloyTriggers.emit(dialogUi.dialog, formCancelEvent);
+            })
+          ]),
+          ...inlineAdditionalBehaviours(editor)
```

Comment:
Use a different function name for easier review and since there is already another function with the same name (different signature) exists,

Reply:
Moved from spread syntax since we don't need it here.